### ISSUE TITLE
[MNG-4165] http session cookies rejected with non-lightweight http wagon (maybe with lightweight one too)

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#5591]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** opened **[MNG-4165](https://issues.apache.org/jira/browse/MNG-4165?redirect=false)** and commented

Reported by Dominic Mitchell on users@maven.apache.org (http://www.nabble.com/Re%3A--PLEASE-TEST--Maven-2.2.0-RC2-p23406267.html)

```
[INFO] Generating "Dependencies" report.
Downloading: http://localhost:8081/nexus/content/groups/public/com/ibm/icu/icu4j/3.4.4/icu4j-3.4.4.jar
Downloading: http://localhost:8081/nexus/content/groups/public/commons-lang/commons-lang/2.1/commons-lang-2.1.jar
May 6, 2009 1:58:26 PM hidden.org.apache.commons.httpclient.HttpMethodBase processCookieHeaders
WARNING: Cookie rejected: "$Version=0; JSESSIONID=E545E65FB5E46552ED8473D17DF1DC80; $Path=/servlets". Illegal path attribute "/servlets". Path of origin: "/nonav/repository//com.google.collections/jars/google-collections-0.9.jar"
May 6, 2009 1:58:30 PM hidden.org.apache.commons.httpclient.HttpMethodBase processCookieHeaders
WARNING: Cookie rejected: "$Version=0; JSESSIONID=C39976B35FDA6AAE7A4C267A7DFF61D3; $Path=/servlets". Illegal path attribute "/servlets". Path of origin: "/nonav/repository//com.ibm.icu/jars/icu4j-3.4.4.jar" 
```

---

**Affects:** 2.2.0

**Issue Links:**
- [WAGON-269](https://issues.apache.org/jira/browse/WAGON-269) Allow configuration of httpclient to change cookie policy, other options

- [WAGON-271](https://issues.apache.org/jira/browse/WAGON-271) Provide configurability of httpclient parameters to allow user to tell Maven to ignore cookies
 (_**"depends upon"**_)

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

This is just a noise issue, at least where Nexus is concerned. It should not affect the ability to resolve or deploy artifacts in any way.

I've committed an improvement to the non-lightweight http wagon in wagon trunk (post 1.0-beta-5 versions), which provides fine-grained configurability in the settings.xml to allow users to tell httpclient to ignore cookies. In this case, since the cookie has the wrong path, and isn't needed anyway, users could safely ignore it. See the linked issue for more details.

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

A fix for this will be provided in wagon 1.0-beta-6, and the solution will be documented with Maven 2.2.0 under http://maven.apache.org/guides/mini/guide-http-settings.html
